tests/test_templater.rs

```diff
@@ -257,6 +257,43 @@ fn test_templater_string_method() {
     insta::assert_snapshot!(render(r#""foo\nbar".first_line()"#), @"foo");
 }
 
+#[test]
+fn test_templater_signature() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    test_env.jj_cmd_success(&repo_path, &["commit", "--message=description"]);
+    let render = |template| get_template_output(&test_env, &repo_path, "@-", template);
+
+    insta::assert_snapshot!(render(r#"author"#), @"Test User <test.user@example.com>");
+    insta::assert_snapshot!(render(r#"author.name()"#), @"Test User");
+    insta::assert_snapshot!(render(r#"author.email()"#), @"test.user@example.com");
+    insta::assert_snapshot!(render(r#"author.username()"#), @"test.user");
+}
+
+#[test]
+fn test_templater_signature_invalid_email() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+    let mut cmd = test_env.jj_cmd(
```

Comment:
I couldn't get this test to take this invalid email, any idea what I'm missing? @martinvonz @yuja

Reply:
No, not really. I think the argument passed as an argument should be ignored, but the `cmd.env()` after looks correct to me.

Reply:
Oh, the `render` closure reads the working-copy commit's parent. Maybe you meant to use `@` instead of `@-` there?

Reply:
Ah, thanks!